workflow-runtime/src/commonMain/kotlin/com/squareup/workflow1/RenderWorkflow.kt

```diff
@@ -135,39 +134,57 @@ public fun <PropsT, OutputT, RenderingT> renderWorkflowIn(
   )
 
   suspend fun <OutputT> sendOutput(
-    actionResult: ActionProcessingResult?,
+    actionResult: ActionProcessingResult,
     onOutput: suspend (OutputT) -> Unit
   ) {
     when (actionResult) {
-      is WorkflowOutput<*> -> {
+      is ActionApplied<*> -> {
         @Suppress("UNCHECKED_CAST")
-        (actionResult as? WorkflowOutput<OutputT>)?.let {
-          onOutput(it.value)
+        (actionResult as? ActionApplied<OutputT>)?.let {
+          it.output?.let { actualOutput ->
+            onOutput(actualOutput.value)
+          }
         }
       }
+
       else -> {} // no -op
     }
   }
 
+  /**
+   * We only check for this when [RuntimeConfig.renderOnStateChangeOnly] is true.
+   */
+  suspend fun maybeCheckNoStateChange(actionResult: ActionProcessingResult): Boolean {
```

Comment:
How about `shortCircuitForUnchangedState`? And have the kdoc explain what should happen when true is returned.

Reply:
"and process another action" seems to imply that we will render, just that we're going to process more actions first. Should this say "and process the next action (if any)"?

Reply:
Just missed me merging this :). But I will follow up with this change.

Reply:
Technically 'processing an action' can involve suspending until there is one but that's more than a little bit confusing.